extensions/resource-deployment/src/services/valueProviderService.ts

```diff
@@ -3,16 +3,24 @@
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import * as vscode from 'vscode';
 import * as rd from 'resource-deployment';
 import * as loc from '../localizedConstants';
 
 class ValueProviderService {
 	private _valueProviderStore = new Map<string, rd.IValueProvider>();
-	registerValueProvider(provider: rd.IValueProvider): void {
+	registerValueProvider(provider: rd.IValueProvider): vscode.Disposable {
 		if (this._valueProviderStore.has(provider.id)) {
 			throw new Error(loc.valueProviderAlreadyDefined(provider.id));
 		}
 		this._valueProviderStore.set(provider.id, provider);
+		return {
+			dispose: () => this.unregisterValueProvider(provider.id)
+		};
+	}
+
+	private unregisterValueProvider(providerId: string): void {
+		this._valueProviderStore.delete(providerId);
```

Comment:
I think it is fine, but could you please confirm it won't throw when providerId doesn't exist in the map?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/delete

```
Return value
true if an element in the Map object existed and has been removed, or false if the element does not exist.
```